presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/InternalResourceGroup.java

```diff
@@ -221,19 +206,34 @@ else if (canQueueMore()) {
     private List<QueryStateInfo> getAggregatedRunningQueriesInfo()
     {
         synchronized (root) {
-            if (subGroups.isEmpty()) {
-                return runningQueries.stream()
-                        .map(QueryExecution::getQueryInfo)
-                        .map(queryInfo -> createQueryStateInfo(queryInfo, Optional.of(id), Optional.empty()))
+            if (!subGroups.isEmpty()) {
+                return subGroups.values().stream()
+                        .map(InternalResourceGroup::getAggregatedRunningQueriesInfo)
+                        .flatMap(List::stream)
                         .collect(toImmutableList());
             }
-            return subGroups.values().stream()
-                    .map(InternalResourceGroup::getAggregatedRunningQueriesInfo)
-                    .flatMap(List::stream)
+
+            List<ResourceGroupInfo> pathToRoot = getPathToRoot();
+
+            return runningQueries.stream()
+                    .map(QueryExecution::getQueryInfo)
+                    .map(queryInfo -> createQueryStateInfo(queryInfo, Optional.of(id), Optional.of(pathToRoot)))
                     .collect(toImmutableList());
         }
     }
 
+    private List<ResourceGroupInfo> getPathToRoot()
+    {
+        ImmutableList.Builder<ResourceGroupInfo> builder = ImmutableList.builder();
```

Comment:
synchronized? Don't want something accidentally call this from out side of a synchronized block accidentally.